### PR TITLE
IGDB Tags integration

### PR DESCRIPTION
**Pull request by W3D3**

### CODE REVIEW

File: src/main/java/io/swagslash/gametrackerserver/igdbconsumer/IGDBWrapper.java [hunk 1]
```diff
@@ -110,19 +113,74 @@ public String getMethod() {
...
         return getCover(game.getCover());
     }
 
-    public List<Cover> getCover(Integer coverId) {
+    public IGDBCover getCover(Integer coverId) {
         IGDBQuery query = new IGDBQuery();
         query.setWhere("id=" + coverId);
-        return getCover(query);
+        List<IGDBCover> covers = getCovers(query);
+        assert covers != null;
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/igdbconsumer/IGDBWrapper.java]: Could be `Objects.requireNonNull(covers)`

Should the application really crash if loading fails?

File: src/main/java/io/swagslash/gametrackerserver/igdbconsumer/Endpoint.java [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 /**
  * Lists all supported endpoints of the IGDB API
- *
+ * currently unused. felt cute, might delete later.
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/igdbconsumer/Endpoint.java:5]: 👍 🤣

File: src/main/java/io/swagslash/gametrackerserver/igdbconsumer/IGDBWrapper.java [hunk 3]
```diff
@@ -110,19 +113,74 @@ public String getMethod() {
...
+            return null;
+        }
+
+        return Arrays.asList(Objects.requireNonNull(result.getBody()));
+    }
+
+    public List<IGDBGameMode> getGameModes(IGDBGame game) {
+        IGDBQuery query = new IGDBQuery();
+
+        String whereString = Arrays.stream(game.getGame_modes())
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/igdbconsumer/IGDBWrapper.java]: Please use camel-case

File: src/main/java/io/swagslash/gametrackerserver/igdbconsumer/model/IGDBCover.java [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-public class Cover {
-    boolean alpha_channel;
-    boolean animated;
-    int game; // The game this cover is associated with
-    int height; // The height of the image in pixels
-    String image_id; // The ID of the image used to construct an IGDB image link
-    String url;
-    int width;
+public class IGDBCover {
+    private boolean alpha_channel;
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/igdbconsumer/model/IGDBCover.java:4]: Please use camel-case
Also below for `image_id`

File: src/main/java/io/swagslash/gametrackerserver/igdbconsumer/IGDBWrapper.java [hunk 5]
```diff
@@ -110,19 +113,74 @@ public String getMethod() {
...
+            return null;
+        }
+
+        return Arrays.asList(Objects.requireNonNull(result.getBody()));
+    }
+
+    public List<IGDBGenre> getGenres(IGDBGame game) {
+        IGDBQuery query = new IGDBQuery();
+
+        String whereString = Arrays.stream(game.getGenres())
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/igdbconsumer/IGDBWrapper.java]: [minor] Duplicate code could be moved to a separate method (see line 153).

File: src/main/java/io/swagslash/gametrackerserver/igdbconsumer/model/IGDBGameMode.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        return slug;
+    }
+
+    public void setSlug(String slug) {
+        this.slug = slug;
+    }
+
+    @Override
+    public String toString() {
+        return "{ GameMode: " + this.name + " }";
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/igdbconsumer/model/IGDBGameMode.java]: [suggestion] Personal preference: `toString` for pure Dtos should include all fields.

File: src/main/java/io/swagslash/gametrackerserver/igdbconsumer/model/IGDBGenre.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        return slug;
+    }
+
+    public void setSlug(String slug) {
+        this.slug = slug;
+    }
+
+    @Override
+    public String toString() {
+        return "{ Genre: " + this.name + " }";
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/igdbconsumer/model/IGDBGenre.java]: [suggestion] Personal preference: `toString` for pure POJOs should include all fields.

**Review by GustavAT — COMMENTED** [review]
LGTM 👍 
Just some minor things

File: src/main/java/io/swagslash/gametrackerserver/igdbconsumer/model/IGDBCover.java (hunk 4 above)
Comment (W3D3) [reply src/main/java/io/swagslash/gametrackerserver/igdbconsumer/model/IGDBCover.java:4]: The IGDB models are mapped to the API, so the variable names will have to stay like this. Don't know if I could change the getter/setter but will look into it.

File: src/main/java/io/swagslash/gametrackerserver/igdbconsumer/IGDBWrapper.java (hunk 1 above)
Comment (W3D3) [reply src/main/java/io/swagslash/gametrackerserver/igdbconsumer/IGDBWrapper.java]: `Objects.requireNonNull` will still throw a null pointer, just explicitly right? So I don't see the benefits.
But yes, the assert is probably overkill.

File: src/main/java/io/swagslash/gametrackerserver/igdbconsumer/model/IGDBGameMode.java (hunk 6 above)
Comment (W3D3) [reply src/main/java/io/swagslash/gametrackerserver/igdbconsumer/model/IGDBGameMode.java]: This is very true, I am just too lazy 😂

**Review by Monomus — APPROVED** [review]

**Pull request merged**